### DISCUSSION TITLE
Adding orWhen in a cycle

### DISCUSSION THREAD

**Discussion by VitaliyMarkelov** [symfony/symfony#58749] [Q&A] [answered]
I have this method in my repository:
```

    public function getByTypes ($types = ['TYPE1', 'TYPE2' , 'TYPE3']): array
    {        

		$queryBuilder = $this->createQueryBuilder('s');
		foreach ($types as $type) {
			$queryBuilder->orWhere('s.field_type = :val')
                        ->setParameter('val', $type);
		}
		return $queryBuilder
                           ->getQuery()
                           ->getResult()
        ;
    }
```

I would expect to get all rows that are of types TYPE1, TYPE2  and TYPE3, but I get only those that mentioned last in the array $types. I have already solved the problem by using andWhere('s.field_type IN (:val)'), but still want to understand why the above code doesn't work.

**Comment by norkunas** [comment, marked as answer]
Because you always override your `:val` parameter and your query in the final just does `where s.field_type = type3 or s.field_type = type3 or s.field_type = type3`

Tip - use `createNamedParameter`:

```php
foreach ($types as $type) {
    $queryBuilder->orWhere('s.field_type = ' .  $qb->createNamedParameter($type));
}
```

**Reply by VitaliyMarkelov** [reply]
Got it, thank you.
